src/app/Recordings/ActiveRecordingsTable.tsx

```diff
@@ -127,15 +124,63 @@ export const ActiveRecordingsTable: React.FunctionComponent<ActiveRecordingsTabl
   }, [addSubscription, context, context.target, refreshRecordingList]);
 
   React.useEffect(() => {
-    merge(
-      context.notificationChannel.messages(NotificationCategory.ActiveRecordingCreated),
-      context.notificationChannel.messages(NotificationCategory.ActiveRecordingStopped),
-      context.notificationChannel.messages(NotificationCategory.ActiveRecordingSaved),
-      context.notificationChannel.messages(NotificationCategory.ActiveRecordingDeleted)
-    ).subscribe(
-      refreshRecordingList
+    addSubscription(
+      combineLatest([
+        context.target.target(),
+        context.notificationChannel.messages(NotificationCategory.ActiveRecordingCreated),
+      ])
+      .subscribe(parts => {
+        const currentTarget = parts[0];
+        const event = parts[1];
+        if (currentTarget.connectUrl != event.message.target) {
+          return;
+        }
```

Comment:
Correct me if I'm wrong, but instead of simply subscribing directly to the `ActiveRecordingCreated` notification category, you've done a `combineLatest` with the current target to ensure that a new recording is only added to the table if the target in the notification matches the current target. Is this to ensure that in the (unlikely?) case that the notification is received _after_ the current target has changed, the active recordings table won't be updated with the wrong value?

Reply:
Ah, good point, I didn't think of other clients.